### DISCUSSION TITLE
Simple playlist system for vidstack?

### DISCUSSION THREAD

**Discussion by AgentSmith0** [vidstack/player#954] [Q&A] [answered]
Is it possible to give vidstack a list of audio files that get played one after another? Also a next and previous button in vidstack would be nice for that.

**Comment by mihar-22** [comment, marked as answer]
That's really easy to do yourself since you can dynamically change the player `src`. It's an array of sources, and `currentSourceIndex` pointer. A prev function will just decrement, and next function will increment. Not sure it's worth adding this to the player.

**Reply by AgentSmith0** [reply]
It it also possible to do this with a `source` element inside the `media-provider` or how can I also set the `type` attribute for the `src` on the `media-player` element?

**Reply by mihar-22** [reply]
You can either set `src` on player itself, or update `<source>` elements inside `<media-provider>`.

```ts
const player = document.querySelector('media-player');

player.src = { src: "...", type: "video/mp4" }
```

**Reply by AgentSmith0** [reply]
The first options works well, thanks!

**Reply by AgentSmith0** [reply]
@mihar-22 Is it also possible to add two custom buttons before and after the play button to navigate through the playlist?

**Reply by mihar-22** [reply]
You’ll need to build your own UI for that.

**Reply by AgentSmith0** [reply]
@mihar-22 Hi, I have the problem, after I change `player.src`, the media does not play when I call `player.play()` or `player.dispatchEvent(new CustomEvent('media-play-request'))`. Is there a way to fix this issue?

**Reply by mihar-22** [reply]
Create a repro and open a new issue so I can look into it.

**Reply by AgentSmith0** [reply]
Done, https://github.com/vidstack/player/issues/1003
